gimie/project.py

```diff
@@ -68,79 +71,41 @@ def __init__(
         else:
             self.project_dir = path
 
-        self.extractors = self.get_extractors(sources)
+        self.extractor = get_extractor(
+            self.url,
+            git_provider,
+            base_url=self.base_url,
+            local_path=self.project_dir,
+        )
+        self.parsers = get_parsers(path, parser_names)
 
     def extract(self) -> Graph:
-        repos = [ex.extract() for ex in self.extractors]
-        graphs = [repo.to_graph() for repo in repos]
-        graph = combine_graphs(*graphs)
-        return graph
+        """Extract repository metadata from git provider to RDF graph and enrich with
+        metadata parsed from file contents."""
 
-    def get_extractors(self, sources: Iterable[str]) -> List[Extractor]:
+        repo = self.extractor.extract()
+        repo_graph = repo.to_graph()
```

Comment:
For the repo, we extract it then make it into a graph. 
For the license, or any parsed file, shouldn't we do the same? 
Suggested process: 
1. The Parser just gets the file/files (see my comment below for file finder vs parser) and returns `SDO.license, URIRef(license_url)`. 
2. Then this gets combined with the uri into a graph (also because for other parsers it would be the same need to combine the extracted information to the uri into a graph, so we could reuse it for other parsers). 
3. Then the parser graph gets combined with the repo graph

Reply:
I see your point, we can indeed use an intermediate tuple representation like `set[tuple[URIRef, URIRef | Literal]]`. 

This would rid Parsers of the unrelated URI and make the `parse()` logic slightly simpler. Let's give it a try!

Reply:
I applied the following changes to enable this:

1. parsers now return a `Set[Tuple[URIRef, Union[URIRef, Literal]]]` as mentioned.
For readability, a type alias is defined in `gimie.graph`:
```
Property: TypeAlias = Tuple[URIRef, Union[URIRef, Literal]]
```
This allows to write signatures as `parse(bytes) -> Set[Property]`

2. A helper function `gimie.graph.operations.properties_to_graph()` helps generating a graph from these tuples.

Since Parsers only need the file contents, they have been decoupled from the `Resource` concept and only take raw data as input. This should make them easier to use in other context.